packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -148,6 +163,7 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
         workInProgress,
         current.child,
         nextChildren,
+        deleteExistingChildren,
```

Comment:
Like discussed offline. Let's call the reconciler twice instead of using boolean flag, first with nextChildren = null, then with current.child = null.